### PR TITLE
Event notification plugins

### LINKED ISSUE #681: Update alert condition and alarm callback plugin documentation for new alerting
The plugin documentation is still showing how to write alert condition and alarm callback plugins. This needs to be updated to event definitions and event notifications:

- http://docs.graylog.org/en/3.1/pages/plugins/alert_conditions.html
- http://docs.graylog.org/en/3.0/pages/plugins/alert_notifications.html

Both APIs should be documented as "unstable" or "alpha" at the moment because they might change in the near future when we are implementing more features.

### PR DESCRIPTION
**Pull request by mariussturm**
I think we need to add some stuff regarding the frontend as well!

Fixes #681

### CODE REVIEW

**Comment by edmundoa** [comment]
I will add the frontend part, then!

File: pages/plugins/event_notifications.rst [hunk 1]
```diff
@@ -0,0 +1,175 @@
+.. _event_notifications_api:
+
+*******************
+Event Notifications
+*******************
+
+Event Notifications are responsible for sending information about events to external systems, such as sending an email, push notifications, opening tickets, writing to chat systems etc.
```
Comment (edmundoa) [inline pages/plugins/event_notifications.rst]:
```suggestion
Event Notifications are responsible for sending information about events to external systems, such as sending an email, push notifications, opening tickets, writing to chat systems, etc.
```

File: pages/plugins/event_notifications.rst [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+  addNotificationType(TYPE_NAME,
+                      EventNotificationConfig.class,
+                      EventNotification.class,
+                      EventNotification.Factory.class);
+
+Example Event Notification
+==========================
+
+You can use the build-in notifications as full examples one for a `configuration class <https://github.com/Graylog2/graylog2-server/blob/3.1/graylog2-server/src/main/java/org/graylog/events/notifications/types/HTTPEventNotificationConfig.java>`_,
```
Comment (edmundoa) [inline pages/plugins/event_notifications.rst]:
```suggestion
You can use the built-in notifications as full examples one for a `configuration class <https://github.com/Graylog2/graylog2-server/blob/3.1/graylog2-server/src/main/java/org/graylog/events/notifications/types/HTTPEventNotificationConfig.java>`_,
```

File: pages/plugins/event_notifications.rst [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+This is called when the scheduler executes the notification and is gathering data for the job.
+
+**public ValidationResult validate()**
+
+On the API level, inputs should be validated, put the checks here.
+
+**public EventNotificationConfigEntity toContentPackEntity(EntityDescriptorIds entityDescriptorIds)**
+
+If content-pack support is desired this methode is called to transfer a notification to a content pack entity.
```
Comment (edmundoa) [inline pages/plugins/event_notifications.rst]:
```suggestion
If content-pack support is desired this method is called to transfer a notification to a content pack entity.
```

File: pages/plugins/event_notifications.rst [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+Take a look at the examples for more details on this.
+
+After creating the configuration class the actual notification code can be written. This is done by implementing the ``EventNotification`` interface in a new class::
+
+  public class HTTPEventNotification implements EventNotification
+
+There is just a single methode to override. The notification logic goes into ``execute(EventNotificationContext ctx)``.
```
Comment (edmundoa) [inline pages/plugins/event_notifications.rst]:
```suggestion
There is just a single method to override. The notification logic goes into ``execute(EventNotificationContext ctx)``.
```

**Review by edmundoa — CHANGES_REQUESTED** [review]
I found a couple typos, other than that looks good to me.

**Review by jalogisch — APPROVED** [review]
👍

**Review by edmundoa — APPROVED** [review]

**Pull request merged**